TTWWW-476: Prevalence Map list styling

mattga11ant opened the pull request:
This PR is for this task: https://simonsfoundation.atlassian.net/browse/TTWWW-476

Styling was the main task, but there were some issues around variables being available on the list page as well as the map page that needed to be worked through.

Four items to note:
1. the design shows this above the list "32 results for administrative database" where "administrative database" is a single filter that could be applied. If there are more filters, would this be a list of filters? That did not seem like a good user experience, so I just updated to "32 results". Figure design can weigh in on this.
2. The existing map has some functionality in the expanded card view on the list page that shows the "confidence interval" with a lot of logic to build these custom circles. The new design does not show this, so assuming this is being removed. I commented on the code for this for now until that is confirmed.
3. Unrelated to this ticket, there was an overalp of style names for the color red, and light-red. I fixed that and also asked Joel to clarify the correct color that should be used for hovers, as in one place the hover is red and in another is is light-red. I updated to be the consistent red for now until he confirms.
4. For the dot color that appears on the SVG map in the list view when expanding the card. The current logic has two colors, a light color and a dark color that changes based on this logic (line 454 in list.ts): if (feature.properties.recommended == 'yes' || feature.properties.recommended == 'Yes'). This is likely something that design did not know. Their design shows the use of the light-red color for this dot which I added as the primary color, and added the dark red for this secondary use case.

- [x] pull this branch
- [x] compile CSS and JS updates
- [x] click on the nav link "LIST" and confirm you see a list view that is consistent with the design in Figma
- [x] confirm that the line above the results that shows the number of results is accurate, and updates when filters are applied
- [x] use the filters to filter results, and confirm the list reflects those filters
- [x] confirm that with filters active, you can switch back and forth between the map and list pages. Both pages should be filtered
- [x] click on a chevron and confirm that the detail card expands and is styled as shown in Figma

yjfnyc commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/joint.ts:
```diff
@@ -2,6 +2,9 @@ import { app } from './app.js';
 export function ttInitJoint() {
     $(document).ready(function () {
+        if (typeof country === 'object') country = '';
+        if (typeof continent === 'object') continent = '';
```
these two lines seem like a hack to me. I did notice when switching from map to list, those two parameters appear as object in the query string. Can you dig a bit to find the cause of it?

yjfnyc commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/list.ts:
```diff
@@ -121,128 +132,141 @@ export function ttInitList() {
-                let mapBlock = card_div.append("div")
-                    .classed("col-4", true);
+                let mapBlock = card_div.append('div')
+                    .classed('test2', true);
```
test2 class doesn't have the definition as far as I know.

mattga11ant replied:
This has been updated.

yjfnyc reviewed (changes requested):
This seems to work well. In terms of functionality, I only see a wording mismatch as shown on the screenshot
<img width="1411" alt="Screenshot 2025-03-09 at 9 27 50 PM" src="https://github.com/user-attachments/assets/9dc7f271-1464-4943-bf15-19a195d16bf4" />

I haven't finish reviewing yet, but would like to give you something to work on so we can work in parallel on this:
1. please see my comments in code.
2. The display of the list has some delay. I can see in production that is also the case. My curiosity is why the map page can appear without the delay as in production. Of course, this is a nice to have, but I would expect this issue would be brought up in QA.

mattga11ant commented:
@yjfnyc I will work to address that hack you mentioned. I was having a heck of a time getting the filters to reset based on the URL variables when switching between the map and list.

I removed that extra test2 class that I forgot.

Can you elaborate on this comment please? "In terms of functionality, I only see a wording mismatch as shown on the screenshot". Are you referring to the missing text "FOR ADMINISTRATIVE DATABASE" that is shown in the design? I made a comment about that in my first PR message. Was curious if we would want to display a comma separated list of all filters?

Regarding the loading delay for the list. I will look into this.

yjfnyc commented:
@mattga11ant let's have a brief chat tomorrow to talk about the issues if you are available. Thanks!

mattga11ant commented:
@yjfnyc I pushed a couple small updates to address the error on the about page and to update the width where the mobile popup appears. For the about page error, we were just loading more JavaScript onto that page than we need.

yjfnyc commented:
Hi @mattga11ant , I brought up the 4 points in our check-in today, and I got the answers from Sam and Joel. Basically they designed it intentionally to not use the visual for confidence and the color distinguishing feature property recommended.

for the copy in your point 1, your current implementation meets their need. No filter related info is needed in it. Thanks!

mattga11ant commented:
@yjfnyc I have addressed all of your comments, including the performance comment. The main item that I found was taking the most time on the list page was creating the map SVG image for every item in the list on page load. I moved this functionality into a function that is now called when a user expands a row. This allows the page to initially load faster, and the time it takes to build a single SVG when a user expands a row is not noticeable. Let me know if this is working well for you as well, or if you would like to revert to the previous method.

yjfnyc reviewed (changes requested):
Hi @mattga11ant , this works really good in terms of functionality. I have the following findings.
1.  The `timeline_type` affecting the filter labels. I think we should implement this here and on the map page. 
2. I also see the unavailable publish link populate the href
3. Cursor shape doesn't need to change when mouse hovering over the dot in the mini-map.
4. variable `thisMapSVG` is not being used in the code.
 
I also have other comments on code. Please see if they make sense to you. Thanks!

mattga11ant commented:
@yjfnyc can you please explain this comment? "The timeline_type affecting the filter labels. I think we should implement this here and on the map page."

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/tailwind.config.js:
```diff
@@ -131,7 +132,10 @@ module.exports = {
+                '1.25': '1.25px',
+                '4': '0.25rem',
                 '6.5': '0.375rem', // 6px
+                '12': '0.75rem', // 12px
```
@yjfnyc This one got off track. Would you like me to go in and update all of these border-widths from 0.5 through 12? And if so, should I make that update as part of this branch?

yjfnyc replied:
@mattga11ant Yes if it wouldn't take too much effort and error prone. In this branch is fine, but should be in a separate commit for easy tracking. Thanks!

mattga11ant replied:
These have been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/list.ts:
```diff
@@ -421,41 +446,47 @@ export function ttInitList() {
-                return "#03C0FF";
+            if (feature.properties.recommended == 'yes' || feature.properties.recommended == 'Yes') {
+                // standard darker red
+                return '#910E1C';
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/list.ts:
```diff
@@ -27,92 +31,164 @@ export function ttInitList() {
+                    .attr('role', 'button')
+                    .attr('aria-expanded', 'false')
+                    .attr('aria-controls', function (d) { 
                         return d.properties.pk; 
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/list.ts:
```diff
@@ -27,92 +31,164 @@ export function ttInitList() {
-                    .classed("fas-red", true)
+                    .classed('border-light-gray2 border-b-1.25', true);
+
+                const toggletd = row1.append('td')
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/list.ts:
```diff
@@ -421,41 +446,47 @@ export function ttInitList() {
                 });
-     
+
+                $('.collapse').on('shown.bs.collapse', function() {
```
This has been updated.

mattga11ant commented:
> Hi @mattga11ant , this works really good in terms of functionality. I have the following findings.
> 
> 1. The `timeline_type` affecting the filter labels. I think we should implement this here and on the map page.
> 2. I also see the unavailable publish link populate the href
> 3. Cursor shape doesn't need to change when mouse hovering over the dot in the mini-map.
> 4. variable `thisMapSVG` is not being used in the code.
> 
> I also have other comments on code. Please see if they make sense to you. Thanks!

@yjfnyc 
1. This has been updated.
2. This was addressed for the map info card, but not on the list page. I added the same functionality to the list page now.
3. This has been updated.
4. This has been removed.

mattga11ant commented:
@yjfnyc this is ready for your re-review. I have addressed each of your comments. I also noticed that there was an error after I merged because the color light-red was re-named to med-red. So I went through and updated all the instances where light-red was still used.

yjfnyc reviewed (approved):
Works well and looks good. Thanks, @mattga11ant
